specs/electra/validator.md

```diff
@@ -189,18 +189,38 @@ def prepare_execution_payload(state: BeaconState,
 
 *[New in Electra]*
 
-1. The execution payload is obtained from the execution engine as defined above using `payload_id`. The response also includes a `execution_requests` entry containing a list of bytes. Each element on the list corresponds to one SSZ list of requests as defined
-in [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). The index of each element in the array determines the type of request.
+1. The execution payload is obtained from the execution engine as defined above using `payload_id`. The response also includes a `execution_requests` entry containing a list of bytes. Each element on the list corresponds to one SSZ list of requests as defined in [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). The first byte of each request is used to determine the request type. There can only be one instance of each request type per execution requests.
 2. Set `block.body.execution_requests = get_execution_requests(execution_requests)`, where:
 
 ```python
-def get_execution_requests(execution_requests: Sequence[bytes]) -> ExecutionRequests:
-    deposits = ssz_deserialize(List[DepositRequest, MAX_DEPOSIT_REQUESTS_PER_PAYLOAD], execution_requests[0])
-    withdrawals = ssz_deserialize(List[WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_PAYLOAD], execution_requests[1])
-    consolidations = ssz_deserialize(List[ConsolidationRequest, MAX_CONSOLIDATION_REQUESTS_PER_PAYLOAD],
-                                     execution_requests[2])
-
-    return ExecutionRequests(deposits, withdrawals, consolidations)
+def get_execution_requests(execution_requests_list: Sequence[bytes]) -> ExecutionRequests:
+    requests = {
+        DEPOSIT_REQUEST_TYPE: {
+            "value": None,
+            "field": "deposits",
+            "type": List[DepositRequest, MAX_DEPOSIT_REQUESTS_PER_PAYLOAD],
+        },
+        WITHDRAWAL_REQUEST_TYPE: {
+            "value": None,
+            "field": "withdrawals",
+            "type": List[WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_PAYLOAD],
+        },
+        CONSOLIDATION_REQUEST_TYPE: {
+            "value": None,
+            "field": "consolidations",
+            "type": List[ConsolidationRequest, MAX_CONSOLIDATION_REQUESTS_PER_PAYLOAD],
+        },
+    }
+
+    execution_requests = ExecutionRequests()
+    for request in execution_requests_list:
+        request_type, request_data = request[0:1], request[1:]
+        assert request_type in requests, "unexpected request type"
```

Comment:
In Lodestar's implementation, we assert `request_type` to be strictly greater than `previous_request_type` due to the ordering requirement as laid out here: https://github.com/ethereum/execution-apis/blob/5f0c46ecf8df0dd4cecf82c49005159c6ed65417/src/engine/prague.md?plain=1#L82

Implemented check at Lodestar [here](https://github.com/ChainSafe/lodestar/blob/70da4a6f6bdd4323821bda7b78fd0783a9f5bc00/packages/beacon-node/src/execution/engine/types.ts#L478)

Reply:
Hmm okay good catch. I will update this PR to do the same. Thank you!